packages/dds/tree/src/test/tablePerformanceTestUtilities.ts

```diff
@@ -0,0 +1,211 @@
+import { createIdCompressor } from "@fluidframework/id-compressor/legacy";
+import {
+	SchemaFactoryAlpha,
+	TreeViewConfiguration,
+	type TreeNodeFromImplicitAllowedTypes,
+	type TreeView,
+} from "../simple-tree/index.js";
+import { TableSchema } from "../tableSchema.js";
+import { DefaultTestSharedTreeKind } from "./utils.js";
+import { AttachState } from "@fluidframework/container-definitions";
+import { MockFluidDataStoreRuntime } from "@fluidframework/test-runtime-utils/internal";
+import { CommitKind, type Revertible } from "../core/index.js";
+import { Tree } from "../shared-tree/index.js";
+import assert from "node:assert";
+
+/**
+ * Define a return type for table tree creation.
+ */
+export interface TableTreeDefinition {
+	/**
+	 * The table tree instance.
+	 */
+	table: TreeNodeFromImplicitAllowedTypes<typeof Table>;
+	/**
+	 * The undo/redo manager for the table tree.
+	 */
+	undoRedoManager: UndoRedoManager;
+}
+
+/**
+ * Factory for creating a table tree schema.
+ * This factory is used to create the schema for the table tree, including cells, columns,
+ * rows, and the table itself.
+ */
+const schemaFactory = new SchemaFactoryAlpha("test");
+
+/**
+ * Defines the schema for a table cell.
+ */
+export const Cell = schemaFactory.string;
+
+/**
+ * Defines the schema for a table column.
+ */
+export class Column extends TableSchema.column({
+	schemaFactory,
+	cell: Cell,
+}) {}
+
+/**
+ * Defines the schema for a table row.
+ */
+export class Row extends TableSchema.row({
+	schemaFactory,
+	cell: Cell,
+}) {}
+
+/**
+ * Defines the schema for a table, which includes columns and rows.
+ * It uses the previously defined Cell, Column, and Row schemas.
+ */
+export class Table extends TableSchema.table({
+	schemaFactory,
+	cell: Cell,
+	column: Column,
+	row: Row,
+}) {}
+
+/**
+ * Provides a simple table tree initialized with the specified size and cell value.
+ * This helper function creates a table schema, initializes a SharedTree instance,
+ * and populates it with the specified number of rows and columns.
+ * Each cell is initialized with the provided cell value.
+ *
+ * @param tableSize - The number of rows and columns to create in the table.
+ * @param cellValue - The initial value to set in each cell of the table.
+ * @returns A fully initialized table tree definition, including table instance, undo/redo stacks, and a cleanup function.
+ */
+export function createTableTree(tableSize: number, cellValue: string): TableTreeDefinition {
+	const sharedTreeFactory = DefaultTestSharedTreeKind.getFactory();
+	const runtime = new MockFluidDataStoreRuntime({
+		idCompressor: createIdCompressor(),
+		attachState: AttachState.Detached,
+	});
+	const tree = sharedTreeFactory.create(runtime, "tree");
+	const treeView = tree.viewWith(
+		new TreeViewConfiguration({
+			schema: Table,
+			enableSchemaValidation: true,
+		}),
+	);
+
+	treeView.initialize(Table.empty());
+	const undoRedoManager = new UndoRedoManager(treeView);
```

Comment:
Initializing this here puts us back in the state where we start building up undo/redo state that we don't necessarily care about (we only really want to track the changes made during `setupOperation`). This also means that _all_ of our tests will incur the cost of undo/redo management even when they don't care about it.

You could always update `runUndoRedoBenchmark` to create and return an `UndoRedoManager` instead - that way only the tests that care about undo/redo incur the cost, and the initialization can happen at the right time in `setupOperation`. You could even let `runUndoRedoBenchmark` handle disposing of this object when the iteration is complete.